pkg/client/actionclient.go

```diff
@@ -90,20 +82,37 @@ func (hcg *actionClientGetter) ActionClientFor(obj client.Object) (ActionInterfa
 	if err != nil {
 		return nil, err
 	}
-	postRenderer := createPostRenderer(rm, actionConfig.KubeClient, obj)
-	return &actionClient{actionConfig, postRenderer}, nil
+	postRenderer := DefaultPostRendererFunc(rm, actionConfig.KubeClient, obj)
+	return &actionClient{
+		conf:               actionConfig,
+		defaultInstallOpts: []InstallOption{WithInstallPostRenderer(postRenderer)},
+		defaultUpgradeOpts: []UpgradeOption{WithUpgradePostRenderer(postRenderer)},
+	}, nil
 }
 
 type actionClient struct {
-	conf         *action.Configuration
+	conf *action.Configuration
+
+	defaultGetOpts       []GetOption
+	defaultInstallOpts   []InstallOption
+	defaultUpgradeOpts   []UpgradeOption
+	defaultUninstallOpts []UninstallOption
+
 	postRenderer postrender.PostRenderer
 }
 
 var _ ActionInterface = &actionClient{}
 
+func concat[T any](s1 []T, s2 ...T) []T {
+	out := make([]T, len(s1))
```

Comment:
As it turns out, I don't think `concat` is necessary at all: https://go.dev/play/p/sqvcMv8lStZ

Reply:
I thought its purpose was to separate the inputs from the outputs to prevent unexpected "action at a distance"?
With pure `append` this can happen depending [in case the first slice has capacity](https://go.dev/play/p/1c2Bh0--rwj).

Reply:
Here the capacity of the `default...` fields should in theory always match their len, so perhaps this is not a concern? WDYT @joelanford ?

Reply:
Ah you're right! I knew there was an edge case here. I think it's best to just go ahead and add the concat back in then. This seems like a potential footgun for our future selves if we don't get this right.

Reply:
Okay latest commit has a re-added concat method that I think has the optimal memory allocation.